src/app/(with-header-sidebar)/dashboard/[id]/components/SettingsColumnModal.tsx

```diff
@@ -0,0 +1,108 @@
+import useModalStore from '@/store/modalStore';
+import { useForm } from 'react-hook-form';
+import Button from '@/components/Button';
+import Input from '@/components/Input';
+import useDashboardStore from '@/store/dashboardStore';
+import { ERROR_MESSAGES } from '@/constants/message';
+import { ColumnFormValue } from '@/types/dashboardView';
+import useDashBoardView from '../hooks/useDashBoardView';
+import { deleteColumn, updateColumn } from '@/lib/columnServie';
+import CloseButton from './CloseButton';
+import styles from './CommonColumnModal.module.css';
+
+export default function SettingsColumnModal({
+  title: prevTitle,
+  id,
+}: ColumnFormValue) {
+  const { closeModal } = useModalStore();
+  const dashboard = useDashboardStore((state) => state.dashboard);
+  const { columns } = useDashBoardView(dashboard ? `${dashboard.id}` : '');
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    watch,
+    formState: { errors },
+  } = useForm<ColumnFormValue>({ mode: 'onChange' });
+
+  const handleError = (message: string) => {
+    setError('title', {
+      type: 'manual',
+      message,
+    });
+  };
+
+  const { title } = watch();
+  const restColumns = columns.filter((column) => column.id !== id);
+  const isDuplicate = restColumns.some((column) => column.title === title);
+
+  const onDelete = async () => {
+    console.log(title);
+    if (!title) {
+      handleError('삭제하려면 해당 컬럼의 이름을 입력해주세요.');
+      return;
+    }
+
+    if (title !== prevTitle) {
+      if (isDuplicate) {
+        handleError('다른 컬럼은 삭제할 수 없습니다.');
+      } else {
+        handleError('존재하지 않는 컬럼입니다.');
+      }
+      return;
+    }
```

Comment:
```suggestion
    if (title === prevTitle) {
      return;
    }

    const errorMessage = isDuplicate 
      ? '다른 컬럼은 삭제할 수 없습니다.' 
      : '존재하지 않는 컬럼입니다.';
      
    handleError(errorMessage);
```
가독성을 높이기 위해 이런 방식도 추천드려요~